src/main/java/org/fenixedu/cms/domain/component/ComponentDescriptor.java

```diff
@@ -74,13 +71,59 @@ public class ComponentDescriptor {
         }
     }
 
+    public static void notNull(Object object, String message) {
+        if (object == null) {
+            throw new IllegalArgumentException(message);
+        }
+    }
+
+    public static Method getMethodIfAvailable(Class<?> clazz, String methodName, Class<?>... paramTypes) {
+        notNull(clazz, "Class must not be null");
+        notNull(methodName, "Method name must not be null");
+        if (paramTypes != null) {
+            try {
+                return clazz.getMethod(methodName, paramTypes);
+            } catch (NoSuchMethodException var9) {
+                return null;
+            }
+        } else {
+            Set<Method> candidates = new HashSet(1);
+            Method[] methods = clazz.getMethods();
```

Comment:
@joaofreitas-qubit sounds like you were decompiling the code directly from the class and stumbled in the java compiler optimizations. If we look into the spring source code for [Classutils](https://github.com/spring-projects/spring-framework/blob/main/spring-core/src/main/java/org/springframework/util/ClassUtils.java#L1518) we see that the implementation is more readable. My suggestion since we are sort of borrowing that from them so we don't need to depend on them to actual have the same code, and also add a reference saying that it was from Classutils from spring